src/buffer/out/TextAttribute.cpp

```diff
@@ -334,9 +352,27 @@ void TextAttribute::SetCrossedOut(bool isCrossedOut) noexcept
 
 void TextAttribute::SetUnderlined(bool isUnderlined) noexcept
 {
-    WI_UpdateFlag(_attrs, CharacterAttributes::Underlined, isUnderlined);
+    if (isUnderlined)
+    {
+        SetUnderlined(UnderlineStyle::SinglyUnderlined);
+    }
+    else
+    {
+        SetUnderlined(UnderlineStyle::NoUnderline);
+    }
+}
+
+void TextAttribute::SetUnderlined(const UnderlineStyle style) noexcept

```

Comment:
I'm not the biggest fan of `SetUnderlined` taking `true` `false` and an enum. Could we rename this one to `SetUnderlineStyle` and just get rid of `SetUnderlined(true)`? All callers would need to be adjusted to specify what style they want, which will clear up some ambiguity :)

Reply:
I am happy to be overruled here. It would be OK to keep it this way too!

Reply:
> Could we rename this one to SetUnderlineStyle and just get rid of SetUnderlined(true)?

Been working on this 😅 👍

And yeah it's better we get rid of `SetUnderlined(bool)`.